### PR TITLE
MD-23 Build esm, common, and umd bundles

### PR DESCRIPTION
**Pull request by joeladam518**
Also:
- Convert to typescript 
- Update eslint config
- Change the build from yarn to npm
- Add auto reviewers

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -15,26 +18,24 @@ jobs:
...
+        run: npm install
 
       - name: Build
-        run: yarn prod
+        run: npm run prod
 
       - name: Publish (NPM)
         run: npm publish --access public
-        env:
-          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (iBotPeaches) [inline .github/workflows/publish.yml]: This was changed to GITHUB_TOKEN and moved to global env scope, but its a real secret for NPM publishing (not GPR) so we should keep that.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -15,26 +18,24 @@ jobs:
...
 
       - uses: actions/setup-node@v4
         with:
+          node-version: ${NODE_VERSION}
+          cache: 'npm'
           registry-url: 'https://npm.pkg.github.com'
 
       - name: Publish (GPR)
         run: npm publish
-        env:
```
Comment (iBotPeaches) [inline .github/workflows/publish.yml]: Likewise here - the reason for not a global `NODE_AUTH_TOKEN` so we could swap it out for a valid token for both npm and google publish.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -1,34 +1,34 @@
...
     steps:
       - uses: actions/checkout@v4
 
-      - name: Setup node
+      - name: Setup Node
         uses: actions/setup-node@v4
         with:
-          node-version: ${{ matrix.node }}
-          cache: 'yarn'
+          node-version: ${NODE_VERSION}
```
Comment (iBotPeaches) [inline .github/workflows/build.yml]: With only 1 usage, I'd just inline personally

File: rollup.config.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
+import ts from 'rollup-plugin-ts';
+import { nodeResolve } from '@rollup/plugin-node-resolve';
+import commonjs from '@rollup/plugin-commonjs';
+import terser from '@rollup/plugin-terser';
+
+const name = 'OnDeckSdk';
```
Comment (iBotPeaches) [inline rollup.config.js]: Nees a rename

File: rollup.config.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    ],
+};
+
+/**
+ * UMD build for projects like ITV & Signage that can't use esm/commonjs modules
+ *
+ * NOTE: As of 2024-03-14, this build is babel-ed to the lowest version of Chrome that most
+ *       itv & signage device can use. Eventually, that might change and we can raise the
+ *       Chrome version.
+ *
```
Comment (iBotPeaches) [inline rollup.config.js]: I'd adjust this comment to avoid project callouts

File: .github/workflows/publish.yml (hunk 1 above)
Comment (joeladam518) [reply .github/workflows/publish.yml]: Thanks. Fixed!

File: .github/workflows/build.yml (hunk 3 above)
Comment (joeladam518) [reply .github/workflows/build.yml]: Thanks. Fixed!

File: rollup.config.js (hunk 5 above)
Comment (joeladam518) [reply rollup.config.js]: Fixed

File: rollup.config.js (hunk 4 above)
Comment (joeladam518) [reply rollup.config.js]: Fixed

File: .github/workflows/publish.yml (hunk 2 above)
Comment (joeladam518) [reply .github/workflows/publish.yml]: Thanks. Fixed!

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -1,34 +1,33 @@
 name: Lint/Test/Build
+env:
+  NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (iBotPeaches) [inline .github/workflows/build.yml]: Do we need the token on this job? The package is public and nothing here appears to publish/require it

File: .github/workflows/publish.yml [hunk 7]
```diff
@@ -32,6 +32,8 @@ jobs:
 
       - uses: actions/setup-node@v4
         with:
+          node-version: 20
+          cache: 'npm'
```
Comment (iBotPeaches) [inline .github/workflows/publish.yml]: I vaguely remembering needing to keep cache off of the 2nd run. Its just used to trick the registry-url to change.

File: .github/workflows/build.yml (hunk 6 above)
Comment (joeladam518) [reply .github/workflows/build.yml]: No it's not needed. I just have got in the habit of adding it. I'll remove.

File: .github/workflows/publish.yml (hunk 7 above)
Comment (joeladam518) [reply .github/workflows/publish.yml]: I'll revert.

File: .github/workflows/build.yml (hunk 6 above)
Comment (joeladam518) [reply .github/workflows/build.yml]: removed

File: .github/workflows/publish.yml (hunk 7 above)
Comment (joeladam518) [reply .github/workflows/publish.yml]: fixed

**Review by iBotPeaches — DISMISSED** [review]

**Review by Wernke96 — DISMISSED** [review]

**Review by xhaferrama — DISMISSED** [review]

**Review by iBotPeaches — COMMENTED** [review]
I tried to fix GitHub CI, but the workflow name (`Lint/Test/Build`) and job name `Lint/Test/Build` are the same - leading to a weird issue.

Can we do like `Lint/Test/Build - Node 20` or something like that?

**Review by stobob — DISMISSED** [review]

File: .github/workflows/publish.yml [hunk 8]
```diff
@@ -15,15 +15,15 @@ jobs:
       - name: Setup node
         uses: actions/setup-node@v4
         with:
-          node-version: 14
-          cache: 'yarn'
-          registry-url: 'https://registry.npmjs.org'
+          node-version: 20
+          cache: 'npm'
+          registry-url: 'https://npm.pkg.github.com'
```
Comment (stobob) [inline .github/workflows/publish.yml]: It doesn't look like this is supposed to change.
```suggestion
          registry-url: 'https://registry.npmjs.org'
```
It should publish to the npm registry and github packages, right?

Comment (joeladam518) [reply .github/workflows/publish.yml]: Good catch. Thanks

**Review by xhaferrama — APPROVED** [review]

**Review by iBotPeaches — APPROVED** [review]

**Review by stobob — APPROVED** [review]

**Pull request merged**